### PR TITLE
Fix upgrade tests

### LINKED ISSUE #73133: Failing test (timeout): [sig-auth] Metadata Concealment should run a check-metadata-concealment job to completion
**Which jobs are failing**:
- [sig-release-master-upgrade#gce-new-master-upgrade-master-parallel](https://testgrid.k8s.io/sig-release-master-upgrade#gce-new-master-upgrade-master-parallel)

**Which test(s) are failing**:
- [[sig-auth] Metadata Concealment should run a check-metadata-concealment job to completion 15m9s](https://gubernator.k8s.io/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-gce-new-master-upgrade-master-parallel/821#sig-auth-metadata-concealment-should-run-a-check-metadata-concealment-job-to-completion)

**Since when has it been failing**:
At least since 2019-01-08

**Testgrid link**:
- [gubernator](https://gubernator.k8s.io/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-gce-new-master-upgrade-master-parallel/821#sig-auth-metadata-concealment-should-run-a-check-metadata-concealment-job-to-completion) 

**Reason for failure**:
Timeout

```
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/auth/metadata_concealment.go:33
Expected error:
    <*errors.errorString | 0xc4200d96b0>: {
        s: "timed out waiting for the condition",
    }
    timed out waiting for the condition
not to have occurred
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/auth/metadata_concealment.go:62
```

### LINKED ISSUE #73134: Failing test in master-upgrade: [sig-network] Networking should provide unchanging, static URL paths for kubernetes api services
**Which jobs are failing**:
- [sig-release-master-upgrade#gce-new-master-upgrade-master-parallel](https://testgrid.k8s.io/sig-release-master-upgrade#gce-new-master-upgrade-master-parallel)

**Which test(s) are failing**:
- [[sig-network] Networking should provide unchanging, static URL paths for kubernetes api services](https://gubernator.k8s.io/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-gce-new-master-upgrade-master-parallel/760#sig-network-networking-should-provide-unchanging-static-url-paths-for-kubernetes-api-services)

**Since when has it been failing**:
2019-01-15 (build [760](https://gubernator.k8s.io/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-gce-new-master-upgrade-master-parallel/760#sig-network-networking-should-provide-unchanging-static-url-paths-for-kubernetes-api-services))

**Testgrid link**:
- [gubernator, build 760](https://gubernator.k8s.io/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-gce-new-master-upgrade-master-parallel/760#sig-network-networking-should-provide-unchanging-static-url-paths-for-kubernetes-api-services)

**Reason for failure**:
```
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/network/networking.go:61
Jan 16 07:18:03.008: Failed: the server could not find the requested resource
```

**Anything else we need to know**:
- The commit combination for the test's first failure is [k/k:51dd616cd](https://github.com/kubernetes/kubernetes/commit/51dd616cd) and [k/test-infra:2d1053361](https://github.com/kubernetes/test-infra/commit/2d1053361). However a few builds prior to #760 have passed with the same combination, and the test has been consistently passing before then in general. Has anything changed outside these repos that could affect the test? Config/setup etc?

/sig network

### PR DESCRIPTION
**Pull request by liggitt**
**What type of PR is this?**
/kind failing-test

**What this PR does / why we need it**:
Two e2e tests are checking behavior that changed in 1.14. This updates those tests to be conditional based on the server version being tested.

**Which issue(s) this PR fixes**:
Fixes #73133
Fixes #73134

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

**Comment by liggitt** [comment]
/cc @dekkagaijin
for metadata concealement

/cc @mbohlool
for swagger schema URL

/cc @feiskyer 
for release branch tagging

/milestone v1.12

**Comment by dekkagaijin** [comment]
LGTM for metadata concealment

**Comment by roycaihw** [comment]
/lgtm

for swagger schema URL

**Review by roycaihw — COMMENTED** [review]
I'm a little surprised that 1.14 changes break 1.12 upgrade test

File: test/e2e/network/networking.go [hunk 1]
```diff
@@ -59,14 +60,21 @@ var _ = SIGDescribe("Networking", func() {
 
 	// First test because it has no dependencies on variables created later on.
 	It("should provide unchanging, static URL paths for kubernetes api services", func() {
+
+		openapiPath := "/swaggerapi"
+		if gte, _ := framework.ServerVersionGTE(utilversion.MustParseSemantic("v1.14.0-alpha.0"), f.ClientSet.Discovery()); gte {
+			framework.Logf("using newer openapi path for 1.14+ server")
+			openapiPath = "/openapi/v2"
+		}
```
[the remark this answers is not in the document]
Comment (liggitt) [reply test/e2e/network/networking.go]: yes, it's just an existence check

**Pull request merged**